routes/products.js

```diff
@@ -36,4 +37,18 @@ router.get("/search", (req, res) => {
   }
 });
 
+router.get("/search/:userId", (req, res) => {
```

Comment:
logic is working please make new file for `routes/posts.js` and include the logic there

mount the router in server js using `app.use("/posts",postRouter)`